src/x86/mod.rs

```diff
@@ -40,15 +42,15 @@ macro_rules! ifunc {
                 } else {
                     fallback::$name as usize
                 };
-            let slot = unsafe { &*(&FN as *const _ as *const AtomicUsize) };
+            let slot = unsafe { &*(FN.get() as *const AtomicUsize) };
```

Comment:
I think the solution is to use `AtomicPtr<()>` (or something else instead of `()`, but it doesn't really matter) instead of `AtomicUsize`.
Then, you might be able to put `AtomicPtr<()>` in the `static` itself, which *is the correct thing to do*.

`static mut` and `static`+`SyncUnsafeCell` are *approximations* of the real `static`+synchronized abstraction.

Reply:
Ah that's perfect. Fn to usize cast was not allowed in statics, but fn to raw pointer is. Thanks!